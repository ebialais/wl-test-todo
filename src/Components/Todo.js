import React from 'react';
import './style.css';

const Todo = (props) => {
  const todo = props.todo
  
  return (
    <li className={ todo.completed ? 'completed todo-element' : 'uncompleted todo-element'}  >
      <p className={ todo.completed ? 'completed-todo-p' : 'todo-p'}>{todo.action}</p>
      <div className='todo-buttons-wrapper'>
        <button
          className='todo-button'
          onClick={()=>props.toggleCompleted(todo.id)}
        >
          <i className="fi fi-br-check"></i>
        </button>
        <button 
          className='todo-button'
          onClick={()=>props.deleteTodo(todo.id)}
        >
          <i className="fi fi-rs-trash"></i>
        </button>
      </div>
    </li>
  )
};

export default Todo;
