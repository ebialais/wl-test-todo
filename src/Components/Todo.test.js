import React from 'react';
import { act } from '@testing-library/react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Todo from './Todo';

test('test uncompleted todo rendering', async () => {
    await act(async () => {
        const props = {todo : { action: 'stuff', completed: false }};
        await act(async () => {
            const renderer = new ShallowRenderer()
            const renderedTodo = renderer.render(<Todo {...props} />);
            expect(renderedTodo).toMatchSnapshot('uncompleted todo');
        });
    });
});

test('test completed todo rendering', async () => {
    await act(async () => {
        const props = {todo : { action: 'stuff', completed: true }};
        await act(async () => {
            const renderer = new ShallowRenderer()
            const renderedTodo = renderer.render(<Todo {...props} />);
            expect(renderedTodo).toMatchSnapshot('completed todo');
        });
    });
});