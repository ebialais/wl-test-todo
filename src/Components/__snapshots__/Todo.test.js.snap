// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test completed todo rendering: completed todo 1`] = `
<li
  className="completed todo-element"
>
  <p
    className="completed-todo-p"
  >
    stuff
  </p>
  <div
    className="todo-buttons-wrapper"
  >
    <button
      className="todo-button"
      onClick={[Function]}
    >
      <i
        className="fi fi-br-check"
      />
    </button>
    <button
      className="todo-button"
      onClick={[Function]}
    >
      <i
        className="fi fi-rs-trash"
      />
    </button>
  </div>
</li>
`;

exports[`test uncompleted todo rendering: uncompleted todo 1`] = `
<li
  className="uncompleted todo-element"
>
  <p
    className="todo-p"
  >
    stuff
  </p>
  <div
    className="todo-buttons-wrapper"
  >
    <button
      className="todo-button"
      onClick={[Function]}
    >
      <i
        className="fi fi-br-check"
      />
    </button>
    <button
      className="todo-button"
      onClick={[Function]}
    >
      <i
        className="fi fi-rs-trash"
      />
    </button>
  </div>
</li>
`;
